lib/src/patterns/observer/Observer.dart

```diff
@@ -107,8 +96,11 @@ class Observer implements IObserver
    * -  Param [Object] - the object to compare.
    * -  Returns [bool] - whether the given object and the [notifyContext] (caller) are the same.
    */
-  bool compareNotifyContext( Object object )
-  {
-       return identical(object, notifyContext);
+  bool compareNotifyContext(Object object) {
+    return identical(object, _notifyContext);
+  }
+
+  set notifyContext(Object? caller) {
+    notifyContext = caller;
```

Comment:
notice here, underscore wasn't added to `notifyContext`, so this setter would behave differently from all the other ones above where it was added. not saying to add it, but to remove it from all other params and vars